pkg/p2p/streamtest/streamtest.go

```diff
@@ -51,8 +53,10 @@ func New(opts ...Option) *Recorder {
 func (r *Recorder) NewStream(_ context.Context, addr swarm.Address, protocolName, protocolVersion, streamName string) (p2p.Stream, error) {
 	recordIn := newRecord()
 	recordOut := newRecord()
-	streamOut := newStream(recordIn, recordOut)
-	streamIn := newStream(recordOut, recordIn)
+	cin := make(chan struct{}, 1)
+	cout := make(chan struct{}, 1)
```

Comment:
No need for buffered channels. Maybe it would be better to call these variables closedIn and closedOut, it would be easier to figure out what they are for right away.